src/main/java/org/sciborgs1155/robot/Constants.java

```diff
@@ -16,11 +27,110 @@
  * @see Units
  */
 public class Constants {
+  /** Returns the robot's alliance. */
+  public static Alliance alliance() {
+    return DriverStation.getAlliance().orElse(Alliance.Blue);
+  }
+
+  /** Returns the rotation of the robot's alliance with respect to the origin. */
+  public static Rotation2d allianceRotation() {
+    return Rotation2d.fromRotations(alliance() == Alliance.Blue ? 0 : 0.5);
+  }
+
   public static final Measure<Time> PERIOD = Seconds.of(0.02); // roborio tickrate (s)
-  public static final double DEADBAND = 0.1;
+  public static final double DEADBAND = 0.15;
   public static final double MAX_RATE =
       DriveConstants.MAX_ACCEL.baseUnitMagnitude()
           / DriveConstants.MAX_ANGULAR_SPEED.baseUnitMagnitude();
-  public static final double SLOW_SPEED = 0.33;
-  public static final double FULL_SPEED = 1.0;
+  public static final double SLOW_SPEED_MULTIPLIER = 0.33;
+  public static final double FULL_SPEED_MULTIPLIER = 1.0;
+
+  // Origin at corner of blue alliance side of field
+  public static class Field {
```

Comment:
the game specific variables in here don't need to exist (basically everything except length and width)